src/instrumentation/initGeneralizationFunctions.ts

```diff
@@ -1,56 +1,78 @@
-import { generalizationFunctionIdentifier } from 'src/app/regionen/[regionSlug]/_mapData/mapDataSources/sources.const'
+import {
+  generalizationFunctionIdentifier,
+  InteracitvityConfiguartion,
+} from 'src/app/regionen/[regionSlug]/_mapData/mapDataSources/sources.const'
 import { prismaClientForRawQueries } from 'src/prisma-client'
-import { Prisma } from '@prisma/client'
 
-// specify license and attribution for data export
-export async function initGeneralizationFunctions(tables) {
+async function createTileSpecification(tableName) {
+  // Get column names and types
+  const columnInformation = await prismaClientForRawQueries.$queryRawUnsafe(`
+  SELECT jsonb_object_agg(column_name, udt_name) - 'geom' - 'minzoom' AS fields
+    FROM information_schema.columns
+    WHERE table_schema = 'public' AND table_name = '${tableName}';`)
+  const { fields } = columnInformation && columnInformation[0] // this object has the form {columnName: columnType}
+  // Get the geometric extent
+  const bbox = await prismaClientForRawQueries.$queryRawUnsafe(
+    `SELECT Array[ST_XMIN(bbox), ST_YMIN(bbox), ST_XMAX(bbox), ST_YMAX(bbox)] as bounds
+    from (
+      SELECT ST_Transform(ST_SetSRID(ST_Extent(geom), 3857), 4326) AS bbox
+        from ${tableName}
+      ) extent;`,
+  )
+  const { bounds } = bbox && bbox[0]
+  // format as vector tile specifaction
+  const tileSpecification = {
+    vector_layers: [{ id: tableName, fields }],
+    bounds,
+  }
+  return tileSpecification
+}
+
+export async function initGeneralizationFunctions(
+  interacitvityConfiguartion: InteracitvityConfiguartion,
+) {
   return Promise.all(
-    tables.map(async (tableName) => {
+    Object.entries(interacitvityConfiguartion).map(async ([tableName, interactivity]) => {
+      // @ts-ignore
```

Comment:
```suggestion
      // @ts-expect-errors
```

wenn immer möglich zu bevorzugen, weile es fehler wird, wenn es mal nicht  mehr nötig ist und weniger ausstrahlt/unterdrückt. 

hier hätte ich aber ein "as string" erwartet.